src/components/common/Tooltip/Tooltip.tsx

```diff
@@ -1,29 +1,31 @@
+import { useState } from 'react';
 import styles from './tooltip.module.scss';
 
 type TooltipProps = {
-    isVisible: boolean;
-    textToShow: string;
-    tooltipPosition: {
-        top?: string;
-        right?: string;
-        bottom?: string;
-        left?: string;
-    };
+    children: React.ReactNode;
+    content: React.ReactNode;
+    tooltipPosition?: React.CSSProperties;
 };
 
-export default function Tooltip({
-    isVisible,
-    textToShow,
-    tooltipPosition,
-}: TooltipProps) {
+const Tooltip = ({ children, content, tooltipPosition }: TooltipProps) => {
+    const [isMouseHovering, setIsMouseHovering] = useState(false);
+    const toggle = () => {
+        setIsMouseHovering((prev) => !prev);
+    };
+
     return (
-        <span
-            data-testid="tooltip"
-            className={`${styles['tooltip']}
-               ${isVisible ? styles['fade-in'] : styles['fade-out']}`}
-            style={{ ...tooltipPosition }}
-        >
-            {textToShow}
+        <span onMouseOver={toggle} onMouseOut={toggle}>
+            {children}
+            <span
+                data-testid="tooltip"
+                className={`${styles['tooltip']}
+        ${isMouseHovering ? styles['fade-in'] : styles['fade-out']}`}
```

Comment:
NIT: this conditional style can be moved out for better readability

Reply:
Removed.